plone/app/caching/operations/etags.py

```diff
@@ -157,12 +157,16 @@ def __init__(self, published, request):
 
     def __call__(self):
         context = getContext(self.published)
-        context_state = queryMultiAdapter(
-            (context, self.request), name="plone_context_state"
-        )
-        if context_state is None:
-            return
-        return "1" if context_state.is_locked() else "0"
+        lock = queryMultiAdapter((context, self.request), name="plone_lock_info")
+
+        if not lock or not lock.lock_info():
```

Comment:
```suggestion
        if not lock:
```

(Avoid calling lock_info() twice)